### PR TITLE
feat: update swift version, macos, and xcode

### PR DESCRIPTION
**Pull request by dayaffe**
Drop support for swift 5.5 and 5.6 and upgrade to min 5.7 and max 5.9 along with associated macos and xcode upgrades.

## Issue \#
#1132 
#1133 

## Description of changes
With the release of Swift 5.9, XCode 15.0, and MacOS 14.0 we need to make sure our SDK supports these new versions. In addition, by dropping support for outdated versions we are able to make use of language features introduced in Swift versions 5.5, 5.6, and 5.7.

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/TestContextGenerator.kt [hunk 1]
```diff
@@ -80,7 +80,7 @@ class TestContextGenerator {
                 .withMember("homepage", Node.from("https://docs.amplify.aws/"))
                 .withMember("author", Node.from("Amazon Web Services"))
                 .withMember("gitRepo", Node.from("https://github.com/awslabs/aws-sdk-swift.git"))
-                .withMember("swiftVersion", Node.from("5.5.0"))
+                .withMember("swiftVersion", Node.from("5.7"))
```
Comment (sichanyoo) [inline codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/TestContextGenerator.kt:83]: Should this be 5.7.0?

**Review by sichanyoo — COMMENTED** [review]
Left some questions in comments.

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -17,29 +17,19 @@ jobs:
...
-          - runner: macos-13
-            xcode: Xcode_13.2.1
-          # Don't run old iOS simulator with new Xcode
-          - destination: 'platform=iOS Simulator,OS=15.2,name=iPhone 13'
-            xcode: Xcode_14.3.1
-          # Don't run new iOS simulator with old Xcode
-          - destination: 'platform=iOS Simulator,OS=16.4,name=iPhone 14'
-            xcode: Xcode_13.2.1
+          - runner: macos-12.5
+            xcode: Xcode_15.0
```
[the remark this answers is not in the document]
Comment (dayaffe) [reply .github/workflows/continuous-integration.yml:32]: We could consider keeping 

```
# Don't run old iOS simulator with new Xcode
- destination: 'platform=iOS Simulator,OS=15.2,name=iPhone 13'
   xcode: Xcode_14.3.1
```

But other than that, the exclusions are related to xcode13 and macos-11 which we are dropping support for. xcode14 runs on minimum macos-12.5

File: AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.swift [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:5.5
+// swift-tools-version:5.7
```
Comment (jbelkins) [inline AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.swift:1]: Similar to smithy-swift, I don't want to change the required version yet.  Only drop to-be-unsupported Swift versions from CI for now.

**Review by jbelkins — CHANGES_REQUESTED** [review]

File: AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.swift (hunk 3 above)
Comment (dayaffe) [reply AWSSDKSwiftCLI/Sources/AWSSDKSwiftCLI/Resources/Package.Base.swift]: `writer.write("// swift-tools-version:${settings.swiftVersion}")`

It looks like swift-tools-version is pinned to our min swiftVersion

**Review by sichanyoo — APPROVED** [review]
Lgtm

**Review by jbelkins — APPROVED** [review]

**Pull request merged**